### ISSUE TITLE
Connections are not disposed when server is closed

### ISSUE THREAD

**Issue by mostroverkhov** [reactor/reactor-netty#495]
### Expected behavior
Connections are disposed on `DisposableServer.disposeNow()`, and `Connection.onDispose()` Publisher is terminated
### Actual behavior
`Connection.onDispose()` Publisher is not terminated

### Steps to reproduce
Code below is timing out

```java
@Test
 public void nettyServerDisposalIssue() {
 MonoProcessor<Void> serverConnDisposed = MonoProcessor.create();

DisposableServer boundServer =
    TcpServer.create()
        .port(8082)
        .doOnConnection(
            serverConnection -> serverConnection.onDispose()
                .subscribe(serverConnDisposed))
        .bind()
        .block();

Connection clientConnection =
    TcpClient.create()
        .addressSupplier(boundServer::address)
        .connect()
        .block();

boundServer.disposeNow();
serverConnDisposed.block(Duration.ofSeconds(5));
```
### Reactor Netty version
0.8.2.RELEASE

### JVM version (e.g. `java -version`)
Java(TM) SE Runtime Environment (build 1.8.0_191-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.191-b12, mixed mode)

### OS version (e.g. `uname -a`)
Linux desktop 4.4.0-138-generic #164~14.04.1-Ubuntu SMP Fri Oct 5 08:56:16 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

**Comment by smaldini** [comment]
Interestingly this is a costly thing to do and thats why Netty doesnt do it by default.
We would need to attach a callback for all connections (and remove them when they are disconnected). I'm not sure this should be a default behavior but something like an option could enable unless we find another algorithm. Would be worth asking Netty team too.

**Comment by smaldini** [comment]
Looking a bit further, Netty will mark the channels as closing if we shutdown the worker loops from the server (which we don't because they can be shared). I wonder if we should force such shutdown  when a server is disposed (and let LoopResources rehydrate new loops for following use).

**Comment by dconnelly** [comment]
Does this mean we expect outstanding connections to be abruptly terminated when a server is disposed? If so, should we consider adding support for graceful shutdown with connection draining?

**Comment by violetagg** [comment]
@dconnelly We plan to shutdown the `EventLoopGroup` invoking `EventExecutorGroup#shutdownGracefully()`. Do you need a specific configuration for the quiet period and the timeout?

**Comment by dconnelly** [comment]
@violetagg This is what I was hoping would happen thanks for the confirmation. Default values (2 second quiet period, 15 second timeout) are fine for my use case at least.

**Comment by violetagg** [comment]
Approach with channel group will be implemented for TCP/HTTP server (as described here for the client):
https://github.com/reactor/reactor-netty/issues/581#issuecomment-457306768

As a workaround for now - `doOnConnection` and `channelGroup.add(connection.channel())` can be used.

This issue will depend on #586

152476f7144e300191b051c5773265a6a5c2f63d  is a test case using `channelGroup` approach

**Comment by violetagg** [comment]
Feature is made available as part of #1022
